app/controllers/theme_controller.py

```diff
@@ -0,0 +1,123 @@
+
+        if theme_data_folder_path.exists():
+            stylesheet_path = theme_data_folder_path
+        elif theme_storage_folder_path.exists():
+            stylesheet_path = theme_storage_folder_path
+        else:
+            # If No Theme is Found, Including Default Theme, Avoid Crash
+            logger.error(
+                f"Stylesheet file including Default Theme not found in '{THEME_FOLDERS}', Applying '{self.theme_name}' Theme"
+            )
+            return ""
+
+        if (
+            stylesheet_path.exists()
+            and stylesheet_path == theme_data_folder_path
+            or theme_storage_folder_path
+        ):
+            return stylesheet_path.read_text()
+        else:
+            # If No Theme is Found, Including Default Theme, Avoid Crash
+            logger.error(
+                f"Stylesheet file including Default Theme not found in '{THEME_FOLDERS}', Applying '{self.theme_name}' Theme"
+            )
+            return ""
+
+    # TODO: Add support for custom icons
+    def theme_icon(self) -> str:
+        theme_data_folder_path = AppInfo().theme_data_folder / self.theme_name / "icons"
+        theme_storage_folder_path = (
+            AppInfo().theme_storage_folder / self.theme_name / "icons"
+        )
+
+        if theme_data_folder_path.exists():
+            icon_folder_path = theme_data_folder_path
+        elif theme_storage_folder_path.exists():
+            icon_folder_path = theme_storage_folder_path
+        else:
+            # If No icon is Found, Including Default Icon, Avoid Crash
+            logger.error(
+                f"Icon folder including Default Icons not found in '{THEME_FOLDERS}', Applying '{self.theme_name}' Icons"
+            )
+            return ""
+
+        if (
+            icon_folder_path.exists()
+            and icon_folder_path == theme_data_folder_path
+            or theme_storage_folder_path
+        ):
+            return icon_folder_path.read_text()
+        else:
+            # If No icon is Found, Including Default Icon, Avoid Crash
+            logger.error(
+                f"Icon folder including Default Icons not found in '{THEME_FOLDERS}', Applying '{self.theme_name}' Icons"
+            )
+            return ""
+
+    @classmethod
+    def get_available_themes(cls) -> list[Path]:
+        """
+        Get a list of available themes from theme data folders.
+        """
+        available_themes = []
+        for theme_path in THEME_FOLDERS:
+            for folder in theme_path.iterdir():
+                if folder.is_dir():
+                    stylesheet_path = folder / "style.qss"
+                    if stylesheet_path.exists():
+                        available_themes.append(folder)
+                    else:
+                        logger.warning(
+                            f"Skipping folder '{folder.name}' in `{theme_path}' as it doesn't contain a valid stylesheet."
+                        )
+
+        return available_themes
```

Comment:
I'm not sure if I agree with this method storing stylesheets. A lot of the themes you see for QT comes as a single file. The differences are differentiated via the filename. The corresponding icons are put in a folder with the same name as the filename.

It'd be a themes folder with all the stylesheets for different themes living at root of the themes folder. The only subfolders in there would be for icons.

As an example, see how MO2 handles it.

Reply:
For structural things that affect how the user would interact with the system, I'd really rather do it now. That avoids potentially breaking user space in the future.

Reply:
all issues related to themes breaking anything are now taken care of